lib/util/http/queue.ts

```diff
@@ -0,0 +1,36 @@
+import URL from 'url';
+import PQueue from 'p-queue';
+import { getRequestLimit } from './host-rules';
+
+const hostQueues = new Map<string | null, PQueue | null>();
+
+function getUrlHost(url: string): string | null {
+  try {
+    return URL.parse(url).host;
+  } catch (e) {
+    return null;
+  }
+}
+
+export function getQueue(url: string): PQueue | null {
+  const host = getUrlHost(url);
```

Comment:
If keys are just hosts, here is possible problem if we're processing multiple repos per run, isn't it?

Reply:
Never mind, we clear queues